src/Hyperion/PrintReport.hs

```diff
@@ -29,6 +28,6 @@ formatReport report =
       | x > 1e3 = show2decs (x/1e3) ++ "us"
       | otherwise = show2decs x ++ "ns"
 
-printReports :: HashMap Text Report -> IO ()
-printReports report = do
-  putDoc $ foldMap formatReport (elems report)
+printReports :: [Report] -> IO ()
```

Comment:
making the result of `printRecords` ordered rather than unordered looks to me like an orthogonal change. Should go in a separate commit, or even in a separate PR insofar as the rest of this PR doesn't build on it.